model/src/pyrenew/model/rtinfectionsrenewal.py

```diff
@@ -292,12 +297,29 @@ def sample(
         )
 
         # Using the predicted infections to sample from the observation process
-        observed, *_ = self.sample_infections_obs(
-            predicted=latent, observed_infections=observed_infections, **kwargs
-        )
+        if self.observation_process is not None:
```

Comment:
Oh I see you're using an if/else for now. I think it's better to take the `NullProcess` approach in the long run, but this is okay for now. I didn't see the equivalent for the admissions model though, so my comment there still applies.